proto/frequenz/api/microgrid/precharge_module.proto

```diff
@@ -0,0 +1,72 @@
+// Contains definitions for precharge modules.
+//
+// Precharging a DC (Direct Current) bus refers to a controlled process in
+// electrical systems, e.g., when connecting an inverter to a battery, where the
+// voltage across a DC bus is gradually increased from a low level to a desired
+// level before connecting a load or initiating normal operation. This process
+// is primarily used to prevent sudden surges of current that can occur when a
+// load is directly connected to a high-voltage DC source.
+//
+// Copyright:
+// Copyright 2023 Frequenz Energy-as-a-Service GmbH
+//
+// License:
+// MIT
+
+syntax = "proto3";
+
+package frequenz.api.microgrid.precharge_module;
+
+import "frequenz/api/microgrid/common.proto";
+
+
+// Enumerated meter states.
+enum ComponentState {
+  // Unspecified state.
+  COMPONENT_STATE_UNSPECIFIED = 0;
+
+  // The precharge and the relsays are opened.
+  COMPONENT_STATE_OFF = 1;
+
+  // Precharging is in progress.
+  COMPONENT_STATE_PRECHARGING = 2;
+
+  // The relays are on/closed, and the precharge circuit is open.
+  COMPONENT_STATE_ON = 3;
+
+  // The precharge module is in an error state.
+  COMPONENT_STATE_ERROR = 4;
+}
+
+// Enumerated error codes.
+enum ErrorCode {
+  // Unspecified.
+  ERROR_CODE_UNSPECIFIED = 0;
+}
+
+// State message.
+message State {
+  // The state of the overall component.
+  ComponentState component_state = 1;
+}
+
+// Error message.
+message Error {
+  // The error code.
+  ErrorCode code = 1;
+
+  // The error severity level.
+  common.ErrorLevel level = 2;
+
+  // The error message.
+  string msg = 3;
+}
+
+// PrechargeModule details
```

Comment:
Same here, though you doing it twice makes me wonder if it's on purpose?

Reply:
Added a `.`.